packages/backend/src/database/migrations/20230309122217_create_org_allowed_email_domains_table.ts

```diff
@@ -0,0 +1,49 @@
+import { Knex } from 'knex';
+
+const OrganizationTableName = 'organizations';
+const OrganizationAllowedEmailDomainsTableName =
+    'organization_allowed_email_domains';
+const organizationMembershipRolesTableName = 'organization_membership_roles';
+
+export async function up(knex: Knex): Promise<void> {
+    if (
+        !(await knex.schema.hasTable(OrganizationAllowedEmailDomainsTableName))
+    ) {
+        await knex.schema.createTable(
+            OrganizationAllowedEmailDomainsTableName,
+            (tableBuilder) => {
+                tableBuilder
+                    .uuid('allowed_email_domains_uuid')
+                    .primary()
+                    .notNullable()
+                    .defaultTo(knex.raw('uuid_generate_v4()'));
+                tableBuilder
+                    .uuid('organization_uuid')
+                    .references('organization_uuid')
+                    .inTable(OrganizationTableName)
+                    .notNullable()
+                    .unique()
```

Comment:
At the moment the UX is for a single configuration, that's why I added the unique.
In the future, if we want to support many-to-one we would need to remove this restriction. Do you think we remove the unique restriction and let the Model handle it? we already do an upsert at the moment based on the org uuid so it would prevent many-to-one.

Reply:
It is ok as it is, but I find it a bit confusing, it looks like it should work if we decide to change the logic  a bit, but then we have this `unique`. In a few months we may not remember why this unique was here, or if we're going to break anything by removing it... If you think it is likely for us to have multiple rules in the short term, I'd say let's remove `unique` . But I have no strong feelings, so it's up to you.